maliput_sim/core/components.py

```diff
@@ -27,19 +27,42 @@
 # OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+class Entity: pass
+class EntityComponentManager: pass
+class SimulationState: pass
```

Comment:
This forward declaration doesn't cope well with type checks (via mypy).

Given that for example the method `def set_entity(self, entity: Entity):` will actually expecting a `maliput_sim.core.components.Entity` when I want it to expect a `maliput_sim.core.ecm.Entity`
Of course, adding an import of that `Entity` in this module will cause a cyclic dependency.

Reply:
I still need to resolve this, by the way, @agalbachicar do you have any experience with this case?

Reply:
I don't have experience with this either. However, I've read the documentation of [type hints](https://peps.python.org/pep-0484/#forward-references) and it seems like this is dealt with a string literal type hint.

Reply:
Oh that's useful, I missed that. Thanks!